secant/Utils/Zatoshi.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  Zatoshi.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 26.05.2022.
+//
+
+import Foundation
+
+struct Zatoshi {
+    enum Constants {
+        static let oneZecInZatoshi: Int64 = 100_000_000
+        static let maxZecs: Int64 = 21_000_000
+        static let maxZatoshi: Int64 = Constants.oneZecInZatoshi * Constants.maxZecs
+    }
+    
+    static var nsDecimalHandler = NSDecimalNumberHandler(
+        roundingMode: NSDecimalNumber.RoundingMode.bankers,
+        scale: 8,
+        raiseOnExactness: true,
+        raiseOnOverflow: true,
+        raiseOnUnderflow: true,
+        raiseOnDivideByZero: true
+    )
+    
+    @Clamped(-Constants.maxZatoshi...Constants.maxZatoshi)
+    var amount: Int64 = 0
+
+    /// Converts `Zatoshi` to `NSDecimalNumber`
+    var asZec: NSDecimalNumber {
+        NSDecimalNumber(decimal: Decimal(amount) / Decimal(Constants.oneZecInZatoshi))
+    }
+
+    /// Converts `Zatoshi` to human readable format, up to 8 fraction digits
+    var asHumanReadableZec: String {
+        asZec.rounding(accordingToBehavior: Zatoshi.nsDecimalHandler).stringValue
+    }
+
+    /// Converts `Decimal` to `Zatoshi`
+    static func from(zec: Decimal) -> Zatoshi {
```

Comment:
Should this receive an NSDecimal or are they Bridged values types?

Reply:
NSDecimal is bridged to Decimal just like NSString is bridged to String